cmd/snap-preseed/main.go

```diff
@@ -0,0 +1,89 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/jessevdk/go-flags"
+)
+
+const (
+	shortHelp = "Prerun the first boot seeding of snaps in a chroot with a snapd seed."
+	longHelp  = `
+The snap-preseed command takes a directory with an image filesytem including
+seed snaps (at /var/lib/snapd/seed) and pre-runs as much as possible without
+booting of first boot snap installation to optimize it.`
+)
```

Comment:
I'd tweak @pedronis words slightly (especially `without booting of first boot snap installation to optimize it`, whose meaning isn't clear to me. Sorry to make another suggestion, but how about:

`The snap-preseed command takes a directory with an image filesytem, including
seed snaps (at /var/lib/snapd/seed), and pre-runs as much as possible to optimise performance without booting the boot snap installation process.`

Reply:
Sadly that rewrite is confusing in different ways,  it sounds we don't start the snap installation process, which is actually what we do. OTOH we don't do any other kind of booting. I need to think a bit.

Reply:
Ok, it's that last part I'm having difficulty with, maybe more like:

`To improves a snap's first-boot caching performance, the snap-preseed command takes a directory with an image filesytem, including seed snaps (at /var/lib/snapd/seed), and pre-runs as much as possible, including much of the boot snap installation process itself.`